include/cosim/algorithm/ecco_algorithm.hpp

```diff
@@ -0,0 +1,97 @@
+/**
+ *  \file
+ *  Defines the class for a ECCO (Energy-Conservation-based Co-Simulation) algorithm
+ *
+ *  \copyright
+ *      This Source Code Form is subject to the terms of the Mozilla Public
+ *      License, v. 2.0. If a copy of the MPL was not distributed with this
+ *      file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+#ifndef LIBCOSIM_ALGORITHM_ECCO_ALGORITHM_HPP
+#define LIBCOSIM_ALGORITHM_ECCO_ALGORITHM_HPP
+
+#include <cosim/algorithm/algorithm.hpp>
+
+namespace cosim
+{
+
+struct ecco_algorithm_params
+{
+    double safety_factor;
+    duration step_size;
+    duration min_step_size;
+    duration max_step_size;
+    double min_change_rate;
+    double max_change_rate;
+    double abs_tolerance;
+    double rel_tolerance;
+    double p_gain;
+    double i_gain;
+};
+
+/**
+ *  A fixed-stepsize co-simulation algorithm.
+ *
+ *  A simple implementation of `algorithm`. The simulation progresses
+ *  at a fixed base stepsize. Simulators are stepped in parallel at an optional
+ *  multiple of this base step size.
+ */
+class ecco_algorithm : public algorithm
+{
+public:
+    /**
+     *  Constructor.
+     *
+     *  \param baseStepSize
+     *      The base communication interval length.
+     *
+     *  \param workerThreadCount
+     *      The number of worker threads to spawn for running FMUs
+     */
+    explicit ecco_algorithm(ecco_algorithm_params params, std::optional<unsigned int> workerThreadCount = std::nullopt);
```

Comment:
FYI, I quickly tried but now i could pass variant type without an error

Reply:
Pass `variant` to where? This constructor? 

TBH I'm not sure if/when variant types can be deduced by the compiler, or if it is always necessary to `std::visit` or `std::get` the specific type.

Reply:
Yep, but I think I am fine to continue with the current version.